File: data/multilang_comparison/program_go/src/func5.c
#include "program_go.h" 
array_t* func5(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array153;
   if (pCounter > 0) {
      array153 = vars->data[--pCounter];
      array153->refC++;
      DEBUG_COPY(array153->id);
   } else {
      array153 = (array_t*)malloc(sizeof(array_t));
      array153->size = 829;
      array153->refC = 1;
      array153->id = 153;
      array153->data = (unsigned int*)malloc(array153->size*sizeof(unsigned int));
      memset(array153->data, 0, array153->size*sizeof(unsigned int));
      DEBUG_NEW(array153->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array153;
   array_t* array154 = func10(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array154->id);
   free(params0.data);
   unsigned int loop45 = 0;
   unsigned int loopLimit45 = (rand()%loopsFactor)/3 + 1;
   for(; loop45 < loopLimit45; loop45++) {
      for (int i = 0; i < array153->size; i++) {
         array153->data[i]++;
      }
   }
   for (int i = 0; i < array154->size; i++) {
      if (array154->data[i] == 69) { 
         return array154;
      }
   }
   array154->refC--;
   if(array154->refC == 0) {
      free(array154->data);
      free(array154);
      DEBUG_FREE(array154->id);
   }
   return array153;
}

File: data/multilang_comparison/program_go/src/func5.go
package main
import "math/rand"
var _ = rand.Intn
func Func5(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array121 *Array
   if pCounter > 0 {
       array121 = vars.Data[pCounter-1]
       pCounter--
       array121.RefC++
   } else {
       array121 = &Array{
           Size: 506,
           RefC: 1,
           Id: 121,
           Data: make([]uint32, 506),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array121
   var array122 *Array = Func10(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop36 int = 0;
   var loopLimit36 int = (50)/3 + 1;
   for ; loop36 < loopLimit36; loop36++ {
      for i := 0; i < array121.Size; i++ {
          array121.Data[i]++
      }
   }
   for i := 0; i < array121.Size; i++ {
       if array121.Data[i] == 28 {
           return array121
       }
   }
   array121.RefC--
   if array121.RefC == 0 {
       array121.Data = nil
       array121 = nil
   }
   return array122;
}

